execution/function/functions.go

```diff
@@ -4,64 +4,29 @@
 package function
 
 import (
-	"fmt"
 	"math"
 	"time"
 
-	"github.com/efficientgo/core/errors"
 	"github.com/prometheus/prometheus/model/histogram"
-
-	"github.com/thanos-io/promql-engine/execution/parse"
-	"github.com/thanos-io/promql-engine/parser"
 )
 
-var InvalidSample = Sample{T: -1, F: 0}
+var invalidSample = sample{T: -1, F: 0}
```

Comment:
Here I would also consider removing `invalidSample` in favour of 3 return arguments and dropping the timestamp.